cmd/localstack/filenotify/filenotify.go

```
@@ -38,14 +38,28 @@ func shouldUseEventWatcher() bool {
 }
 
 // New tries to use a fs-event watcher, and falls back to the poller if there is an error
-func New(interval time.Duration) (FileWatcher, error) {
+func New(interval time.Duration, fileWatcherStrategy string) (FileWatcher, error) {
+	if fileWatcherStrategy != "" {
+		log.Debugln("Forced usage of filewatcher strategy: ", fileWatcherStrategy)
+		if fileWatcherStrategy == "event" {
+			if watcher, err := NewEventWatcher(); err == nil {
+				return watcher, nil
+			} else {
+				log.Fatalln("Event based filewatcher is selected, but unable to start. Please try setting the filewatcher to polling. Error: ", err)
+			}
+		} else if fileWatcherStrategy == "polling" {
+			return NewPollingWatcher(interval), nil
+		} else {
+			log.Fatalf("Invalid filewatcher strategy %s. Only event and polling are allowed.\n", fileWatcherStrategy)
```

Would it make sense to crash the init binary rather than just log here?
Currently, it just hangs until a timeout kills the container :(

---

A fatal log should indeed crash the init binary

---

I see.
Then it's likely one of the error cases we do not (yet) handle actively in the init. Reporting back to LocalStack would be nice in the future to short-circuit the timeout (which is likely high when debugging).